packages/monaco/src/browser/monaco-editor-provider.ts

```diff
@@ -105,6 +105,10 @@ export class MonacoEditorProvider {
 
     protected async getModel(uri: URI, toDispose: DisposableCollection): Promise<MonacoEditorModel> {
         const reference = await this.textModelService.createModelReference(uri);
+        // if document is invalid makes sure that all events from underlying resource are processed before throwing invalid model
+        if (!reference.object.valid) {
+            await reference.object.sync();
+        }
```

Comment:
This change was not part of the PR originally; can you please give us some details why you've made this change? Is this related to the flaky tests or does something with the embedded code editor part? Thanks!

Reply:
It was added to fix an issue revealed by tests: https://github.com/eclipse-theia/theia/pull/7539#issuecomment-613444764

> Dirty editor tests should be good now too. Code was assuming that we always load editor model to open from the disk or already processed all disk changes. But tests revealed that there are cases when we did not receive all events yet but already trying to open an editor. I've added a guard that if a model looks invalid then sync with the disk before throwing an invalid model exception.

Reply:
👍 Thank you!